src/Controller/CarController.php

```diff
@@ -171,30 +172,40 @@ public function show(#[ValueResolver(CarValueResolver::class)]
             new OA\Response(response: 400, description: 'Invalid input data')
         ]
     )]
-        public function new(Request $request,EventDispatcherInterface $eventDispatcher): Response
+    public function new(Request $request, EventDispatcherInterface $eventDispatcher): Response
     {
-        $car = new Car();
         $user = $this->security->getUser();
 
         if (!$user) {
             return $this->redirectToRoute('app_login');
         }
 
-        $car->setUser($user);
-        $form = $this->createForm(CarFormType::class, $car);
+        $form = $this->createForm(CarFormType::class, null);
 
         $form->handleRequest($request);
-        
+
         if ($form->isSubmitted() && $form->isValid()) {
-            $this->carService->createNewCar($car);  
-        
+            $carData = $form->getData();
+
+            $car = $this->carFactory->create(
```

Comment:
move car factory in service, don't call it directly in controller. That's why you have service layer